pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/process/ProcessRuntime.java

```diff
@@ -106,6 +106,10 @@
         switch (instanceConfig.getFunctionDetails().getRuntime()) {
             case JAVA:
                 logConfigFile = "java_instance_log4j2.xml";
+                String palsarHome = System.getenv("PULSAR_HOME");
```

Comment:
Please also move this line under the subsequent "if" statement as it is only used when the "if" conditional resolves to true